tests/python_package_test/test_sklearn.py

```diff
@@ -617,6 +617,79 @@ def test_nan_handle(self):
         gbm = lgb.LGBMRegressor(**params).fit(**params_fit)
         np.testing.assert_array_equal(gbm.evals_result_['training']['l2'], np.nan)
 
+    def test_first_metric_only(self):
+        X, y = load_boston(True)
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+
+        params = {'n_estimators': 25,
+                  'learning_rate': 0.8,
+                  'num_leaves': 15,
+                  'verbose': -1,
+                  'seed': 123}
+        params_fit = {'X': X_train,
+                      'y': y_train,
+                      'eval_set': (X_train, y_train),
+                      'early_stopping_rounds': 5,
+                      'verbose': False}
+
+        def fit_and_check(eval_set_name, metric_names, assumed_iteration, first_metric_only):
+            params['first_metric_only'] = first_metric_only
+            gbm = lgb.LGBMRegressor(**params).fit(**params_fit)
+            self.assertIn(eval_set_name, gbm.evals_result_)
+            for metric_name in metric_names:
+                self.assertIn(metric_name, gbm.evals_result_[eval_set_name])
+                self.assertEqual(len(gbm.evals_result_[eval_set_name][metric_name]),
+                                 assumed_iteration + (params_fit['early_stopping_rounds']
+                                                      if eval_set_name != 'training'
+                                                      else params['n_estimators']))
+                self.assertEqual(assumed_iteration, gbm.best_iteration_)
+
+        # training data as eval_set
+        fit_and_check('training', ['l2'], 0, False)
+        fit_and_check('training', ['l2'], 0, True)
+
+        # single eval_set
+        params_fit['eval_set'] = (X_test, y_test)
+        fit_and_check('valid_0', ['l2'], 2, False)
+        fit_and_check('valid_0', ['l2'], 2, True)
+
+        params_fit['eval_metric'] = "l2"
+        fit_and_check('valid_0', ['l2'], 2, False)
+        fit_and_check('valid_0', ['l2'], 2, True)
+
+        params_fit['eval_metric'] = "l1"
+        fit_and_check('valid_0', ['l1', 'l2'], 2, False)
+        fit_and_check('valid_0', ['l1', 'l2'], 8, True)
+
+        params_fit['eval_metric'] = ["l1", "l2"]
+        fit_and_check('valid_0', ['l1', 'l2'], 2, False)
+        fit_and_check('valid_0', ['l1', 'l2'], 8, True)
+
+        params_fit['eval_metric'] = ["l2", "l1"]
+        fit_and_check('valid_0', ['l1', 'l2'], 2, False)
+        fit_and_check('valid_0', ['l1', 'l2'], 2, True)
+
+        # two eval_set
```

Comment:
@matsuken92  I'm sorry for my question, but why is this group of tests needed? It seems that the only difference compared to `# single eval_set` group is `valid_1` instead of `valid_0` name. The case for training data is not used for early stopping is checked in `# training data as eval_set` group of tests.

Reply:
At first, this bug was found the additional function `eval_train_metric`. By the influence, train dataset is coming to `_callback()` of early_stopping and the training dataset was used to check whether stop or not. Then does not stopped in early stopping.
So I think it is important to check wide variety of the combination of eval_set , then just in case I add these test items. However, theoretically no problem if these patterns are removed. So I can remove `# two eval_set` part.

Reply:
@matsuken92 So, the aim of `# two eval_set` is the same as of `# training data as eval_set`, right? Then, I think we can decrease the number of checks in this group to save CI time. I mean, we don't need different cases of `params_fit['eval_metric']` to check that training data is not used in early stopping. Instead, there should be good to add a case with two real eval sets (valid_0 and valid_1) and check that all of them is used in early stopping. 
> Requires at least one validation data and one metric.
            If there's more than one, will check all of them. But the training data is ignored anyway.

WDYT?

Reply:
@StrikerRUS 
Yes, I modified `# two eval_set` part to use split into two real eval datasets.